### PR TITLE
Multiple recordings

### PR DESCRIPTION
**Pull request by StevesBro**
useSelector and useDispatch hooks have been implemented with Redux for AudioRecorder.tsx

Todo: Actually fix the issue where the recorder is breaking when recording a second time.

---
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/thecombine/439)

### CODE REVIEW

File: src/components/Pronunciations/AudioRecorder.tsx [hunk 1]
```diff
@@ -7,7 +7,9 @@ import {
 } from "@material-ui/core";
 import { red } from "@material-ui/core/colors";
 import FiberManualRecord from "@material-ui/icons/FiberManualRecord";
-import React, { useState } from "react";
+import React from "react";
```
Comment (languagewarrior) [inline src/components/Pronunciations/AudioRecorder.tsx]: Duplicate of #

- [ ] ###  @****

File: src/components/Pronunciations/AudioRecorder.tsx [hunk 2]
```diff
@@ -7,7 +7,9 @@ import {
...
+import { useSelector, useDispatch } from "react-redux";
+import { updateRecordingStatus } from "../../goals/ReviewEntries/ReviewEntriesComponent/ReviewEntriesActions";
```
Comment (languagewarrior) [inline src/components/Pronunciations/AudioRecorder.tsx]: import { updateRecordingStatus } from "../../goals/ReviewEntries/ReviewEntriesComponent/ReviewEntriesActions";

File: src/components/Pronunciations/AudioRecorder.tsx [hunk 3]
```diff
@@ -42,7 +44,10 @@ function getFileNameForWord(wordId: string): string {
 }
 
 export default function AudioRecorder(props: RecorderProps) {
-  const [isRecording, setIsRecording] = useState(false);
+  const isRecording = useSelector(
+    (state: any) => state.reviewEntriesState.isRecording
+  );
+  const dispatch = useDispatch();
```
Comment (languagewarrior) [inline src/components/Pronunciations/AudioRecorder.tsx]: import { updateRecordingStatus } from "../../goals/ReviewEntries/ReviewEntriesComponent/ReviewEntriesActions";

File: src/components/Pronunciations/AudioRecorder.tsx [hunk 4]
```diff
@@ -51,17 +56,18 @@ export default function AudioRecorder(props: RecorderProps) {
   function safeStartRecording(
     event: Event | React.TouchEvent | React.MouseEvent
   ) {
+    console.log(isRecording);
```
Comment (languagewarrior) [inline src/components/Pronunciations/AudioRecorder.tsx]: console.log(isRecording);

File: src/components/Pronunciations/AudioRecorder.tsx [hunk 5]
```diff
@@ -51,17 +56,18 @@ export default function AudioRecorder(props: RecorderProps) {
...
     if (!isRecording) {
+      dispatch(updateRecordingStatus(true));
```
Comment (languagewarrior) [inline src/components/Pronunciations/AudioRecorder.tsx]: console.log(isRecording);
    console.log(isRecording);
      dispatch(updateRecordingStatus(true));

File: src/components/Pronunciations/AudioRecorder.tsx [hunk 6]
```diff
@@ -51,17 +56,18 @@ export default function AudioRecorder(props: RecorderProps) {
...
       recorder.startRecording();
-      setIsRecording(true);
     }
   }
 
   function safeStopRecording(
     event: Event | React.TouchEvent | React.MouseEvent
   ) {
-    if (isRecording) {
+    if (/*isRecording*/ true) {
```
Comment (languagewarrior) [inline src/components/Pronunciations/AudioRecorder.tsx]: if (/*isRecording*/ true) {

File: src/components/Pronunciations/AudioRecorder.tsx [hunk 7]
```diff
@@ -84,7 +90,7 @@ export default function AudioRecorder(props: RecorderProps) {
           // <Translate id="pronunciations.noMicAccess" />;
           // TODO: Show alert dialog here
         })
-        .finally(() => setIsRecording(false));
+        .finally(() => dispatch(updateRecordingStatus(false)));
```
Comment (languagewarrior) [inline src/components/Pronunciations/AudioRecorder.tsx]: .finally(() => dispatch(updateRecordingStatus(false)));

File: src/goals/ReviewEntries/ReviewEntriesComponent/ReviewEntriesActions.tsx [hunk 8]
```diff
@@ -58,6 +58,13 @@ function updateWord(
   };
 }
 
+export function updateRecordingStatus(recordingStatus: boolean) {
+  return {
+    type: ReviewEntriesActionTypes.UpdateRecordingStatus,
+    recordingStatus,
+  };
+}
+
```
Comment (languagewarrior) [inline src/goals/ReviewEntries/ReviewEntriesComponent/ReviewEntriesActions.tsx]: .finally(() => dispatch(updateRecordingStatus(false)));
export function updateRecordingStatus(recordingStatus: boolean) {
   return {
     type: ReviewEntriesActionTypes.UpdateRecordingStatus,
     recordingStatus,
   };
 }

File: src/components/Pronunciations/AudioRecorder.tsx (hunk 2 above)
Comment (languagewarrior) [inline src/components/Pronunciations/AudioRecorder.tsx]: import { useSelector, useDispatch } from "react-redux";
  import { updateRecordingStatus } from "../../goals/ReviewEntries/ReviewEntriesComponent/ReviewEntriesActions";

File: src/components/Pronunciations/AudioRecorder.tsx (hunk 1 above)
Comment (languagewarrior) [reply src/components/Pronunciations/AudioRecorder.tsx]: mport { useSelector, useDispatch } from "react-redux";
  import { updateRecordingStatus } from "../../goals/ReviewEntries/ReviewEntriesComponent/ReviewEntriesActions";

**Comment by imnasnainaec** [comment]
Resolves #407

**Pull request closed without merging**